flink-connectors/flink-connector-elasticsearch-base/src/main/java/org/apache/flink/streaming/connectors/elasticsearch/table/ElasticsearchOptions.java

```diff
@@ -0,0 +1,132 @@
+	public static final ConfigOption<String> FAILURE_HANDLER_OPTION =
+		ConfigOptions.key("failure-handler")
+			.stringType()
+			.defaultValue("fail")
+			.withDescription(Description.builder()
+				.text("Failure handling strategy in case a request to Elasticsearch fails")
+				.list(
+					text("\"fail\" (throws an exception if a request fails and thus causes a job failure),"),
+					text("\"ignore\" (ignores failures and drops the request),"),
+					text("\"retry_rejected\" (re-adds requests that have failed due to queue capacity saturation),"),
+					text("\"class name\" for failure handling with a ActionRequestFailureHandler subclass"))
+				.build());
+	public static final ConfigOption<Boolean> FLUSH_ON_CHECKPOINT_OPTION =
+		ConfigOptions.key("sink.flush-on-checkpoint")
+			.booleanType()
+			.defaultValue(true)
+			.withDescription("Disables flushing on checkpoint");
+	public static final ConfigOption<Integer> BULK_FLUSH_MAX_ACTIONS_OPTION =
+		ConfigOptions.key("sink.bulk-flush.max-actions")
+			.intType()
+			.noDefaultValue()
+			.withDescription("Maximum number of actions to buffer for each bulk request.");
+	public static final ConfigOption<MemorySize> BULK_FLASH_MAX_SIZE_OPTION =
+		ConfigOptions.key("sink.bulk-flush.max-size")
+			.memoryType()
+			.noDefaultValue()
+			.withDescription("Maximum size of buffered actions per bulk request");
+	public static final ConfigOption<Duration> BULK_FLUSH_INTERVAL_OPTION =
+		ConfigOptions.key("sink.bulk-flush.interval")
+			.durationType()
+			.noDefaultValue()
+			.withDescription("Bulk flush interval");
+	public static final ConfigOption<BackOffType> BULK_FLUSH_BACKOFF_TYPE_OPTION =
+		ConfigOptions.key("sink.bulk-flush.backoff.type")
+			.enumType(BackOffType.class)
+			.defaultValue(BackOffType.DISABLED)
+			.withDescription("Backoff strategy");
+	public static final ConfigOption<Integer> BULK_FLUSH_BACKOFF_MAX_RETRIES_OPTION =
+		ConfigOptions.key("sink.bulk-flush.backoff.max-retries")
+			.intType()
+			.noDefaultValue()
+			.withDescription("Maximum number of retries.");
+	public static final ConfigOption<Duration> BULK_FLUSH_BACKOFF_DELAY_OPTION =
+		ConfigOptions.key("sink.bulk-flush.backoff.delay")
+			.durationType()
+			.noDefaultValue()
+			.withDescription("Delay between each backoff attempt.");
+	public static final ConfigOption<Duration> CONNECTION_MAX_RETRY_TIMEOUT_OPTION =
+		ConfigOptions.key("connection.max-retry-timeout")
+			.durationType()
+			.noDefaultValue()
+			.withDescription("Maximum timeout between retries.");
+	public static final ConfigOption<String> CONNECTION_PATH_PREFIX =
+		ConfigOptions.key("connection.path-prefix")
+			.stringType()
+			.noDefaultValue()
+			.withDescription("Prefix string to be added to every REST communication.");
+	public static final ConfigOption<String> FORMAT_OPTION =
+		ConfigOptions.key("format")
+			.stringType()
+			.defaultValue("json")
```

Comment:
I think we should add a validation on the format value? Should throw exception to tell users ES only supports json?

Reply:
I thought about it. I decided not to do that. It would really be a weak check. We can validate only the `factoryIdentifier` of the format. It still does not guarantee it is out `json` format.

Moreover I thought that if we do not add the validation, we give users more flexibility to override the format with their own version. The only requirement on the format is that it produces a valid json document. It does not have to use our format.

Nevertheless I can add that validation, if you want.

Reply:
Makes sense to me.